PyInstaller/depend/bytecode.py

```diff
@@ -81,30 +83,80 @@ def finditer(pattern: Pattern, string):
+        ((?:`EXTENDED_ARG`.)*
+         (?:`LOAD_NAME`|`LOAD_GLOBAL`|`LOAD_FAST`).)
+
+        # For foo.bar.whizz(), the above is the 'foo', below is the 'bar.whizz'.
+        ((?:(?:`EXTENDED_ARG`.)*
+         (?:`LOAD_METHOD`|`LOAD_ATTR`).)*)
+
+        # Load however many arguments it takes. These (for now) must all be constants.
+        # Again, code with >256 constants may need extended enumeration.
+        ((?:(?:`EXTENDED_ARG`.)*
+         `LOAD_CONST`.)*)
+
+        # Call the function. The parameter is the argument count (which may also be >256) if CALL_FUNCTION or
+        # CALL_METHOD are used. For CALL_FUNCTION_EX, the parameter are flags.
+        ((?:`EXTENDED_ARG`.)*
+         (?:`CALL_FUNCTION`|`CALL_METHOD`|`CALL_FUNCTION_EX`).)
+    """
+    )
+else:
+    # Starting with python 3.11, the bytecode is peppered with CACHE instructions (which dis module conveniently hides
+    # unless show_caches=True is used). Dealing with these CACHE instructions in regex rules is going to render them
+    # unreadable, so instead we pre-process the bytecode and filter the offending opcodes out.
+    def _cleanup_code(code):
+        CACHE = dis.opmap["CACHE"]
+
+        out = []
+        for idx in range(0, len(code), 2):
+            if code[idx] == CACHE:
+                continue
+            out.append(code[idx])
+            out.append(code[idx+1])
+
+        return bytes(out)
+
+    # Python 3.11 removed CALL_FUNCTION and CALL_METHOD, and replaced them with PRECALL + CALL instructions.
+    # The CALL_FUNCTION_EX is still present.
+
+    # language=PythonVerboseRegExp
+    _call_function_bytecode = bytecode_regex(
+        rb"""
+        # Matches `global_function('some', 'constant', 'arguments')`.
+
+        # Load the global function. In code with >256 of names, this may require extended name references.
+        ((?:`EXTENDED_ARG`.)*
+         (?:`LOAD_NAME`|`LOAD_GLOBAL`|`LOAD_FAST`).)
+
+        # For foo.bar.whizz(), the above is the 'foo', below is the 'bar.whizz'.
+        ((?:(?:`EXTENDED_ARG`.)*
+         (?:`LOAD_METHOD`|`LOAD_ATTR`).)*)
+
+        # Load however many arguments it takes. These (for now) must all be constants.
+        # Again, code with >256 constants may need extended enumeration.
+        ((?:(?:`EXTENDED_ARG`.)*
+         `LOAD_CONST`.)*)
+
+        # Call the function. Python 3.11 removed CALL_FUNCTION and CALL_METHOD in favor of PRECALL + CALL; both
+        # have the same argument (argc = argument count), so for our purposes, it suffices to match just up to the
+        # PRECALL instruction (which also saves us from having to worry about CACHE instructions between PRECALL and
+        # CALL). The argument count may probably be > 256 for PRECALL/CALL. For CALL_FUNCTION_EX, the parameter
+        # are flags.
+        ((?:`EXTENDED_ARG`.)*
+         (?:`CALL_FUNCTION_EX`|`PRECALL`).)
+    """
+    )
```

Comment:
Perhaps a better way of dealing with these Python version dependent op codes would be to alias them out in `_instruction_to_regex()` like we already do for the two already in that function? Any op-codes that don't have an equivalent can return an empty character set (`b"[]"`) to no-op them. Then we should be able to stick to one definition of `_call_function_bytecode`.

Reply:
Hmm, I'm not particularly fond of that remapping in `_instruction_to_regex`. I think it is going to get very confusing really fast, because for pre-3.7 (which we could remove now, I suppose), we have to map CALL_METHOD to CALL_FUNCTION, but in post-3.11, we'll need to map CALL_METHOD and CALL_FUNCTION to PRECALL.

I think a better (more intuitive and easier to read) approach is to have a single `_call_function_bytecode` regex, into which the four matching groups (global function, function/method, args, call) are injected from symbolic constants. These constants could then be defined on per-version basis, and would directly spell out which opcodes are available there. 

For example, the "call" group for pre-3.7 would be `` `CALL_FUNCTION`|`CALL_FUNCTION_EX` ``, for post-3.7 `` CALL_FUNCTION`|`CALL_METHOD`|`CALL_FUNCTION_EX` ``, and for post-3.11 `` `PRECALL`|`CALL_FUNCTION_EX` ``. 

We'd need to do the same at least for function/method part (due to LOAD_METHOD being unavailable in pre-3.7), but for consistency sake, I think all four groups should be declared in a separate variable and injected into the final regex.

Reply:
Yeah, let's go for that (including removing the remapping already in `_instruction_to_regex()`).